### PR TITLE
Update TensorFlow plugin documentation

### PR DESCRIPTION
**Pull request by JanuszL**
- adds a proper reference to DALIRawIterator inside DALIIterator documentation
- lists the default values of the arguments for DALIDataset operator

Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

#### Why we need this PR?
*Pick one, remove the rest*
- It updates TensorFlow plugin documentation

#### What happened in this PR?
*Fill relevant points, put NA otherwise. Replace anything inside []*
 - What solution was applied:
     adds a proper reference to DALIRawIterator inside DALIIterator documentation
     lists the default values of the arguments for DALIDataset operator
 - Affected modules and functionalities:
     TF plugin documentation
 - Key points relevant for the review:
     wording
 - Validation and testing:
     CI
 - Documentation (including examples):
     only the documentation has been updated

Relates to https://github.com/NVIDIA/DALI/issues/2326

**JIRA TASK**: *[NA]*

### CODE REVIEW

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali/python/nvidia/dali/plugin/tf.py [hunk 1]
```diff
@@ -42,9 +42,11 @@ def serialize_pipeline(pipeline):
 def DALIIteratorWrapper(pipeline = None, serialized_pipeline = None, sparse = [],
                         shapes = [], dtypes = [], batch_size = -1, prefetch_queue_depth = 2, **kwargs):
   """
-TF Plugin Wrapper
+  TF Plugin Wrapper
 
-This operator works in the same way as DALI TensorFlow plugin, with the exception that is also accepts Pipeline objects as the input and serializes it internally. For more information, please look **TensorFlow Plugin API reference** in the documentation.
+  This operator works in the same way as DALI TensorFlow plugin, with the exception that is also
+  accepts Pipeline objects as the input and serializes it internally. For more information,
```
Comment (jantonguirao) [inline dali/python/nvidia/dali/plugin/tf.py:48]:
```suggestion
  accepts Pipeline objects as an input, which are serialized internally. For more information,
```

File: dali/python/nvidia/dali/plugin/tf.py [hunk 2]
```diff
@@ -42,9 +42,11 @@ def serialize_pipeline(pipeline):
...
+  please look :meth:`nvidia.dali.plugin.tf.DALIRawIterator` in the documentation.
```
Comment (jantonguirao) [inline dali/python/nvidia/dali/plugin/tf.py:49]:
```suggestion
  see :meth:`nvidia.dali.plugin.tf.DALIRawIterator`.
```

File: dali/python/nvidia/dali/plugin/tf.py [hunk 3]
```diff
@@ -176,6 +178,8 @@ def __init__(
       self._pipeline = serialize_pipeline(pipeline)
       self._batch_size = batch_size
       self._num_threads = num_threads
+      if device_id is None:
```
Comment (jantonguirao) [inline dali/python/nvidia/dali/plugin/tf.py:181]:
```suggestion
      device_id = device_id or types.CPU_ONLY_DEVICE_ID
```

File: dali_tf_plugin/daliop.cc [hunk 4]
```diff
@@ -86,7 +86,7 @@ REGISTER_OP("Dali")
   .Doc(R"doc(
 DALI TensorFlow plugin
 
-Creates a Dali pipeline for classification tasks from serialized DALI pipeline (given in `serialized_pipeline` parameter).
+Creates a Dali pipeline from serialized DALI pipeline (given in `serialized_pipeline` parameter).
```
Comment (jantonguirao) [inline dali_tf_plugin/daliop.cc:89]:
```suggestion
Creates a DALI pipeline from a serialized pipeline, obtained from `serialized_pipeline` argument.
```

File: docs/installation.rst [hunk 5]
```diff
@@ -160,13 +160,7 @@ Weekly builds
...
 
 Also, there is a weekly release channel with more thorough testing. To access most recent weekly
-builds please use flowing release channel:
-
-* for CUDA 10:
-
-.. code-block:: bash
-
-  pip install --extra-index-url https://developer.download.nvidia.com/compute/redist/weekly nvidia-dali-weekly-cuda100 nvidia-dali-tf-plugin-weekly-cuda100
+builds please use flowing release channel (available only for CUDA 11):
```
Comment (jantonguirao) [inline docs/installation.rst:163]:
```suggestion
builds please use the following release channel (available only for CUDA 11):
```
If you already say it's only available for CUDA 11, I am not sure if line 165 still applies

Comment (JanuszL) [reply docs/installation.rst]: Removed

File: dali_tf_plugin/daliop.cc (hunk 4 above)
Comment (JanuszL) [reply dali_tf_plugin/daliop.cc]: Done

File: dali/python/nvidia/dali/plugin/tf.py (hunk 3 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: Done

File: dali/python/nvidia/dali/plugin/tf.py (hunk 2 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: Done

File: dali/python/nvidia/dali/plugin/tf.py (hunk 1 above)
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: Done

File: dali/python/nvidia/dali/plugin/tf.py [hunk 6]
```diff
@@ -42,9 +42,11 @@ def serialize_pipeline(pipeline):
 def DALIIteratorWrapper(pipeline = None, serialized_pipeline = None, sparse = [],
                         shapes = [], dtypes = [], batch_size = -1, prefetch_queue_depth = 2, **kwargs):
   """
-TF Plugin Wrapper
+  TF Plugin Wrapper
 
-This operator works in the same way as DALI TensorFlow plugin, with the exception that is also accepts Pipeline objects as the input and serializes it internally. For more information, please look **TensorFlow Plugin API reference** in the documentation.
+  This operator works in the same way as DALI TensorFlow plugin, with the exception that is also
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: Done

**Comment by JanuszL** [comment]
!build

**Comment by JanuszL** [comment]
!build

File: dali_tf_plugin/daliop.cc [hunk 7]
```diff
@@ -86,7 +86,7 @@ REGISTER_OP("Dali")
   .Doc(R"doc(
 DALI TensorFlow plugin
 
-Creates a Dali pipeline for classification tasks from serialized DALI pipeline (given in `serialized_pipeline` parameter).
+Creates a DDALI pipeline from a serialized pipeline, obtained from `serialized_pipeline` argument.
```
Comment (jantonguirao) [inline dali_tf_plugin/daliop.cc:89]:
```suggestion
Creates a DALI pipeline from a serialized pipeline, obtained from `serialized_pipeline` argument.
```

**Review by jantonguirao — APPROVED** [review]

File: dali_tf_plugin/daliop.cc (hunk 7 above)
Comment (JanuszL) [reply dali_tf_plugin/daliop.cc]: Done

File: dali/python/nvidia/dali/plugin/tf.py [hunk 8]
```diff
@@ -176,6 +178,7 @@ def __init__(
       self._pipeline = serialize_pipeline(pipeline)
       self._batch_size = batch_size
       self._num_threads = num_threads
+      device_id = device_id or types.CPU_ONLY_DEVICE_ID
```
Comment (klecki) [inline dali/python/nvidia/dali/plugin/tf.py:181]: This doesn't allow to use `0` as device_id, as it will always set the second part of or statement.

Comment (klecki) [reply dali/python/nvidia/dali/plugin/tf.py:181]:
```suggestion
      device_id = device_id if device_id is not None else types.CPU_ONLY_DEVICE_ID
```
Is this the intended outcome?

Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: Right. I tested how it work only for None and non-0 value.

Comment (JanuszL) [reply dali/python/nvidia/dali/plugin/tf.py]: I will keep the old way as it is more readable than one line combo.
```
if device_id is None:
  device_id = types.CPU_ONLY_DEVICE_ID
```

**Comment by JanuszL** [comment]
!build

**Review by klecki — APPROVED** [review]

**Pull request merged**